File: src/routes/resume/_skills.ts
export interface Skill {
  name: string;
  level: "Expert" | "Fluent" | "Amateur" | "Basics";
}

export default [
  // Programming Languages
  { name: "JavaScript/Node.js", level: "Expert" },
  { name: "TypeScript", level: "Expert" },
  { name: "Golang", level: "Expert" },
  { name: "Rust", level: "Fluent" },
  { name: "Python", level: "Fluent" },
  { name: "Ruby/Ruby on Rails", level: "Expert" },
  { name: "Java", level: "Fluent" },
  { name: "C/C++", level: "Fluent" },
  { name: "Haskell", level: "Fluent" },
  { name: "BASH", level: "Fluent" },
  { name: "Perl", level: "Amateur" },
  { name: ".NET", level: "Amateur" },
  { name: "Erlang", level: "Basics" },
  { name: "Smalltalk", level: "Basics" },
  { name: "Visual Works", level: "Basics" },
  { name: "AmberJS", level: "Basics" },
  { name: "LiveScript", level: "Fluent" },

  // Web Technologies
  { name: "React", level: "Expert" },
  { name: "GraphQL", level: "Expert" },
  { name: "REST APIs", level: "Expert" },
  { name: "Webpack", level: "Expert" },
  { name: "Babel", level: "Expert" },
  { name: "MobX", level: "Fluent" },
  { name: "Redux", level: "Fluent" },
  { name: "Relay", level: "Fluent" },

  // Databases & Data
  { name: "PostgreSQL", level: "Expert" },
  { name: "MySQL", level: "Expert" },
  { name: "SQL", level: "Expert" },
  { name: "DynamoDB", level: "Fluent" },
  { name: "Graph Databases", level: "Fluent" },
  { name: "Graph Data Structures", level: "Amateur" },
  { name: "BigQuery", level: "Fluent" },
  { name: "Apache NiFi", level: "Fluent" },

  // Cloud & Infrastructure
  { name: "Kubernetes", level: "Expert" },
  { name: "Docker", level: "Expert" },
  { name: "AWS", level: "Expert" },
  { name: "Google Cloud Platform", level: "Expert" },
  { name: "Terraform", level: "Fluent" },
  { name: "Packer", level: "Fluent" },
  { name: "Pulumi", level: "Fluent" },
  { name: "CloudFormation", level: "Fluent" },
  { name: "kops", level: "Fluent" },
  { name: "AWS CodePipeline", level: "Fluent" },
  { name: "AWS CodeBuild", level: "Fluent" },
  { name: "Google Cloud Build", level: "Fluent" },
  { name: "Google PubSub", level: "Fluent" },

  // DevOps & CI/CD
  { name: "Git", level: "Expert" },
  { name: "Mercurial", level: "Fluent" },
  { name: "Version Control", level: "Expert" },
  { name: "Continuous Deployment", level: "Expert" },
  { name: "CI/CD", level: "Expert" },
  { name: "Build Pipelines", level: "Expert" },

  // Blockchain & Cryptocurrency
  { name: "Blockchain", level: "Expert" },
  { name: "Cosmos", level: "Expert" },
  { name: "Bitcoin", level: "Fluent" },
  { name: "Cryptocurrency", level: "Fluent" },
  { name: "Distributed Systems", level: "Expert" },
  { name: "Genesis Ceremony", level: "Expert" },

  // System Administration
  { name: "Linux", level: "Expert" },
  { name: "Mac OS X", level: "Expert" },
  { name: "BSD", level: "Expert" },
  { name: "Windows", level: "Fluent" },
  { name: "VMware", level: "Fluent" },
  { name: "System Administration", level: "Expert" },
  { name: "DNS", level: "Fluent" },
  { name: "Email Systems", level: "Fluent" },
  { name: "Intrusion Detection", level: "Fluent" },

  // Business Intelligence & Analytics
  { name: "Business Objects BI", level: "Fluent" },
  { name: "Data Analytics", level: "Fluent" },
  { name: "SEO", level: "Fluent" },
  { name: "Performance Optimization", level: "Expert" },

  // Communication & Protocols
  { name: "SIP", level: "Fluent" },
  { name: "VoIP", level: "Fluent" },
  { name: "Network Protocols", level: "Fluent" },
  { name: "Message Queues", level: "Fluent" },
  { name: "RabbitMQ", level: "Fluent" },

  // Architecture & Design
  { name: "Microservices", level: "Expert" },
  { name: "System Architecture", level: "Expert" },
  { name: "API Design", level: "Expert" },
  { name: "Database Design", level: "Expert" },
  { name: "Scalability", level: "Expert" },
  { name: "Performance Tuning", level: "Expert" },
  { name: "Infrastructure as Code", level: "Expert" },

  // Payment & Financial
  { name: "Payment Processing", level: "Fluent" },
  { name: "Braintree", level: "Fluent" },
  { name: "Financial Systems", level: "Fluent" },
  { name: "Transaction Processing", level: "Fluent" },

  // Testing & Quality
  { name: "Testing Frameworks", level: "Fluent" },
  { name: "Blockchain Testing", level: "Expert" },
  { name: "Performance Testing", level: "Fluent" },
  { name: "Automated Testing", level: "Expert" },

  // Leadership & Soft Skills
  { name: "Technical Leadership", level: "Expert" },
  { name: "Team Management", level: "Expert" },
  { name: "Mentoring", level: "Expert" },
  { name: "Cross-functional Collaboration", level: "Expert" },
  { name: "Problem Solving", level: "Expert" },
  { name: "Architecture Planning", level: "Expert" },
  { name: "Project Management", level: "Fluent" },
  { name: "Customer Satisfaction", level: "Expert" },
  { name: "Entrepreneurship", level: "Fluent" },
  { name: "Consulting", level: "Expert" },

  // Languages
  { name: "English", level: "Expert" },
  { name: "Spanish", level: "Basics" },
] as Skill[];

File: yew/src/data/skills.rs
use crate::models::{Skill, SkillLevel};

pub fn get_skills() -> Vec<Skill> {
    vec![
        // Programming Languages
        Skill {
            name: "JavaScript/Node.js".to_string(),
            level: SkillLevel::Expert,
        },
        Skill {
            name: "TypeScript".to_string(),
            level: SkillLevel::Expert,
        },
        Skill {
            name: "Golang".to_string(),
            level: SkillLevel::Expert,
        },
        Skill {
            name: "Rust".to_string(),
            level: SkillLevel::Fluent,
        },
        Skill {
            name: "Python".to_string(),
            level: SkillLevel::Fluent,
        },
        Skill {
            name: "Ruby/Ruby on Rails".to_string(),
            level: SkillLevel::Expert,
        },
        Skill {
            name: "Java".to_string(),
            level: SkillLevel::Fluent,
        },
        Skill {
            name: "C/C++".to_string(),
            level: SkillLevel::Fluent,
        },
        Skill {
            name: "Haskell".to_string(),
            level: SkillLevel::Fluent,
        },
        Skill {
            name: "BASH".to_string(),
            level: SkillLevel::Fluent,
        },
        Skill {
            name: "Perl".to_string(),
            level: SkillLevel::Fluent,
        },
        Skill {
            name: ".NET".to_string(),
            level: SkillLevel::Fluent,
        },
        Skill {
            name: "Erlang".to_string(),
            level: SkillLevel::Fluent,
        },
        Skill {
            name: "Smalltalk".to_string(),
            level: SkillLevel::Fluent,
        },
        Skill {
            name: "Visual Works".to_string(),
            level: SkillLevel::Fluent,
        },
        Skill {
            name: "AmberJS".to_string(),
            level: SkillLevel::Fluent,
        },
        Skill {
            name: "LiveScript".to_string(),
            level: SkillLevel::Fluent,
        },
        // Web Technologies
        Skill {
            name: "React".to_string(),
            level: SkillLevel::Expert,
        },
        Skill {
            name: "GraphQL".to_string(),
            level: SkillLevel::Expert,
        },
        Skill {
            name: "REST APIs".to_string(),
            level: SkillLevel::Expert,
        },
        Skill {
            name: "Webpack".to_string(),
            level: SkillLevel::Expert,
        },
        Skill {
            name: "Babel".to_string(),
            level: SkillLevel::Expert,
        },
        Skill {
            name: "MobX".to_string(),
            level: SkillLevel::Fluent,
        },
        Skill {
            name: "Redux".to_string(),
            level: SkillLevel::Fluent,
        },
        Skill {
            name: "Relay".to_string(),
            level: SkillLevel::Fluent,
        },
        // Databases & Data
        Skill {
            name: "PostgreSQL".to_string(),
            level: SkillLevel::Expert,
        },
        Skill {
            name: "MySQL".to_string(),
            level: SkillLevel::Expert,
        },
        Skill {
            name: "SQL".to_string(),
            level: SkillLevel::Expert,
        },
        Skill {
            name: "DynamoDB".to_string(),
            level: SkillLevel::Fluent,
        },
        Skill {
            name: "Graph Databases".to_string(),
            level: SkillLevel::Fluent,
        },
        Skill {
            name: "Graph Data Structures".to_string(),
            level: SkillLevel::Fluent,
        },
        Skill {
            name: "BigQuery".to_string(),
            level: SkillLevel::Fluent,
        },
        Skill {
            name: "Apache NiFi".to_string(),
            level: SkillLevel::Fluent,
        },
        // Cloud & Infrastructure
        Skill {
            name: "Kubernetes".to_string(),
            level: SkillLevel::Expert,
        },
        Skill {
            name: "Docker".to_string(),
            level: SkillLevel::Expert,
        },
        Skill {
            name: "AWS".to_string(),
            level: SkillLevel::Expert,
        },
        Skill {
            name: "Google Cloud Platform".to_string(),
            level: SkillLevel::Expert,
        },
        Skill {
            name: "Terraform".to_string(),
            level: SkillLevel::Fluent,
        },
        Skill {
            name: "Packer".to_string(),
            level: SkillLevel::Fluent,
        },
        Skill {
            name: "Pulumi".to_string(),
            level: SkillLevel::Fluent,
        },
        Skill {
            name: "CloudFormation".to_string(),
            level: SkillLevel::Fluent,
        },
        Skill {
            name: "kops".to_string(),
            level: SkillLevel::Fluent,
        },
        Skill {
            name: "AWS CodePipeline".to_string(),
            level: SkillLevel::Fluent,
        },
        Skill {
            name: "AWS CodeBuild".to_string(),
            level: SkillLevel::Fluent,
        },
        Skill {
            name: "Google Cloud Build".to_string(),
            level: SkillLevel::Fluent,
        },
        Skill {
            name: "Google PubSub".to_string(),
            level: SkillLevel::Fluent,
        },
        // DevOps & CI/CD
        Skill {
            name: "Git".to_string(),
            level: SkillLevel::Expert,
        },
        Skill {
            name: "Mercurial".to_string(),
            level: SkillLevel::Fluent,
        },
        Skill {
            name: "Version Control".to_string(),
            level: SkillLevel::Expert,
        },
        Skill {
            name: "Continuous Deployment".to_string(),
            level: SkillLevel::Expert,
        },
        Skill {
            name: "CI/CD".to_string(),
            level: SkillLevel::Expert,
        },
        Skill {
            name: "Build Pipelines".to_string(),
            level: SkillLevel::Expert,
        },
        // Blockchain & Cryptocurrency
        Skill {
            name: "Blockchain".to_string(),
            level: SkillLevel::Expert,
        },
        Skill {
            name: "Cosmos".to_string(),
            level: SkillLevel::Expert,
        },
        Skill {
            name: "Bitcoin".to_string(),
            level: SkillLevel::Fluent,
        },
        Skill {
            name: "Cryptocurrency".to_string(),
            level: SkillLevel::Fluent,
        },
        Skill {
            name: "Distributed Systems".to_string(),
            level: SkillLevel::Expert,
        },
        Skill {
            name: "Genesis Ceremony".to_string(),
            level: SkillLevel::Expert,
        },
        // System Administration
        Skill {
            name: "Linux".to_string(),
            level: SkillLevel::Expert,
        },
        Skill {
            name: "Mac OS X".to_string(),
            level: SkillLevel::Expert,
        },
        Skill {
            name: "BSD".to_string(),
            level: SkillLevel::Expert,
        },
        Skill {
            name: "Windows".to_string(),
            level: SkillLevel::Fluent,
        },
        Skill {
            name: "VMware".to_string(),
            level: SkillLevel::Fluent,
        },
        Skill {
            name: "System Administration".to_string(),
            level: SkillLevel::Expert,
        },
        Skill {
            name: "DNS".to_string(),
            level: SkillLevel::Fluent,
        },
        Skill {
            name: "Email Systems".to_string(),
            level: SkillLevel::Fluent,
        },
        Skill {
            name: "Intrusion Detection".to_string(),
            level: SkillLevel::Fluent,
        },
        // Business Intelligence & Analytics
        Skill {
            name: "Business Objects BI".to_string(),
            level: SkillLevel::Fluent,
        },
        Skill {
            name: "Data Analytics".to_string(),
            level: SkillLevel::Fluent,
        },
        Skill {
            name: "SEO".to_string(),
            level: SkillLevel::Fluent,
        },
        Skill {
            name: "Performance Optimization".to_string(),
            level: SkillLevel::Expert,
        },
        // Communication & Protocols
        Skill {
            name: "SIP".to_string(),
            level: SkillLevel::Fluent,
        },
        Skill {
            name: "VoIP".to_string(),
            level: SkillLevel::Fluent,
        },
        Skill {
            name: "Network Protocols".to_string(),
            level: SkillLevel::Fluent,
        },
        Skill {
            name: "Message Queues".to_string(),
            level: SkillLevel::Fluent,
        },
        Skill {
            name: "RabbitMQ".to_string(),
            level: SkillLevel::Fluent,
        },
        // Architecture & Design
        Skill {
            name: "Microservices".to_string(),
            level: SkillLevel::Expert,
        },
        Skill {
            name: "System Architecture".to_string(),
            level: SkillLevel::Expert,
        },
        Skill {
            name: "API Design".to_string(),
            level: SkillLevel::Expert,
        },
        Skill {
            name: "Database Design".to_string(),
            level: SkillLevel::Expert,
        },
        Skill {
            name: "Scalability".to_string(),
            level: SkillLevel::Expert,
        },
        Skill {
            name: "Performance Tuning".to_string(),
            level: SkillLevel::Expert,
        },
        Skill {
            name: "Infrastructure as Code".to_string(),
            level: SkillLevel::Expert,
        },
        // Payment & Financial
        Skill {
            name: "Payment Processing".to_string(),
            level: SkillLevel::Fluent,
        },
        Skill {
            name: "Braintree".to_string(),
            level: SkillLevel::Fluent,
        },
        Skill {
            name: "Financial Systems".to_string(),
            level: SkillLevel::Fluent,
        },
        Skill {
            name: "Transaction Processing".to_string(),
            level: SkillLevel::Fluent,
        },
        // Testing & Quality
        Skill {
            name: "Testing Frameworks".to_string(),
            level: SkillLevel::Fluent,
        },
        Skill {
            name: "Blockchain Testing".to_string(),
            level: SkillLevel::Expert,
        },
        Skill {
            name: "Performance Testing".to_string(),
            level: SkillLevel::Fluent,
        },
        Skill {
            name: "Automated Testing".to_string(),
            level: SkillLevel::Expert,
        },
        // Leadership & Soft Skills
        Skill {
            name: "Technical Leadership".to_string(),
            level: SkillLevel::Expert,
        },
        Skill {
            name: "Team Management".to_string(),
            level: SkillLevel::Expert,
        },
        Skill {
            name: "Mentoring".to_string(),
            level: SkillLevel::Expert,
        },
        Skill {
            name: "Cross-functional Collaboration".to_string(),
            level: SkillLevel::Expert,
        },
        Skill {
            name: "Problem Solving".to_string(),
            level: SkillLevel::Expert,
        },
        Skill {
            name: "Architecture Planning".to_string(),
            level: SkillLevel::Expert,
        },
        Skill {
            name: "Project Management".to_string(),
            level: SkillLevel::Fluent,
        },
        Skill {
            name: "Customer Satisfaction".to_string(),
            level: SkillLevel::Expert,
        },
        Skill {
            name: "Entrepreneurship".to_string(),
            level: SkillLevel::Fluent,
        },
        Skill {
            name: "Consulting".to_string(),
            level: SkillLevel::Expert,
        },
        // Languages
        Skill {
            name: "English".to_string(),
            level: SkillLevel::Expert,
        },
        Skill {
            name: "Spanish".to_string(),
            level: SkillLevel::Fluent,
        },
    ]
}

pub fn get_skills_by_level(level: SkillLevel) -> Vec<Skill> {
    get_skills()
        .into_iter()
        .filter(|skill| skill.level == level)
        .collect()
}
